### PR TITLE
add len method to Arena and fix #14

### LINKED ISSUE #14: Avoid integer overflows for large requests
You check if a buffer doesn't have space using `len + new > capacity`. The left hand side could overflow. If you instead check `new > capactity - len` you avoid this issue.

replace

    if chunks.current.len() + iter_min_len > chunks.current.capacity() {

by

    if iter_min_len > chunks.current.capacity() - chunks.current.len() {

and

    if chunks.current.len() + num > chunks.current.capacity() {

by

    if num > chunks.current.capacity() - chunks.current.len() {

The second one of these might even lead to a buffer overflow.

### LINKED ISSUE #16: add `len` method
This is useful for using the size of previous typed arenas to build new typed arenas with large enough spaces.

### PR DESCRIPTION
**Pull request by Zaechus**
I added a len method for Arena.

Fixes #16
Fixes #14

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -201,7 +201,8 @@ impl<T> Arena<T> {
 
         let iter_min_len = iter.size_hint().0;
         let mut next_item_index;
-        if chunks.current.len() + iter_min_len > chunks.current.capacity() {
+        // if chunks.current.len() + iter_min_len > chunks.current.capacity() {
+        if iter_min_len > chunks.current.capacity() - chunks.current.len() {
```
Comment (fitzgen) [inline src/lib.rs]: Care to also add

```rust
debug_assert!(
    chunks.current.capacity() >= chunks.current.len(),
    "capacity is always greater than or equal to len, so we don't need to worry about underflow"
);
```

here and below? Thanks.

File: src/test.rs [hunk 2]
```diff
@@ -77,6 +79,7 @@ fn test_zero_cap() {
     let b = arena.alloc(2);
     assert_eq!(*a, 1);
     assert_eq!(*b, 2);
+    assert_eq!(arena.len(), 2);
```
Comment (fitzgen) [inline src/test.rs]: Can you also add a test that exercises `len` when the arena has multiple chunks? Thanks.

Comment (Maxgy) [reply src/test.rs]: I believe I do this in the test `arena_as_intended`. Is this enough or is more needed?

Comment (fitzgen) [reply src/test.rs]: Ah yeah I missed that that test was exercising that path. Just the other debug assertion and then we can merge this, then.

**Comment by Maxgy** [comment]
I think it's good to go.

**Pull request merged**